samples/azure-quantum/variational-quantum-eigensolver/VQE-qiskit-hydrogen-ionq-sim.ipynb

```diff
@@ -0,0 +1,520 @@
+    "optimizer = SPSA(maxiter=100)\n",
+    "var_form = EfficientSU2(qubitOp.num_qubits, entanglement=\"linear\")\n",
+    "vqe = VQE(qubitOp, var_form, optimizer=optimizer)\n",
+    "ret = vqe.run(quantum_instance)\n",
+    "vqe_result = np.real(ret['eigenvalue'] + molecule.nuclear_repulsion_energy)\n",
+    "print(\"VQE Result on noisy simulator:\", vqe_result)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8a7dd86d",
+   "metadata": {},
+   "source": [
+    "The parameters found by the optimization loop:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "57b42add",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "array([-0.02589263, -3.57591366, -4.02171876,  3.30633207, -1.5098052 ,\n",
+       "        3.75248329, -2.08578439,  3.38280164,  3.96292972, -4.9768716 ,\n",
+       "       -2.92184533,  1.01813741, -2.59110068, -2.37222154,  3.35072313,\n",
+       "       -2.33565366])"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "p0 = ret.optimal_point\n",
+    "p0"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e19dd20c",
+   "metadata": {},
+   "source": [
+    "The energy was evaluated a total of `ret.cost_function_evals` times until the minimum was found."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "1d273554",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "241"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "ret.cost_function_evals"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4517f80e",
+   "metadata": {},
+   "source": [
+    "### Circuit visualization\n",
+    "\n",
+    "Each energy evaluation consists of two circuits that are run. You can visualize these circuits with Qiskit using the `vqe` instance."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "c247b443",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Remove idle wires\n",
+    "# https://quantumcomputing.stackexchange.com/questions/25672/remove-inactive-qubits-from-qiskit-circuit\n",
+    "from qiskit.converters import circuit_to_dag, dag_to_circuit\n",
+    "from collections import OrderedDict\n",
+    "\n",
+    "def remove_idle_qwires(circ):\n",
+    "    dag = circuit_to_dag(circ)\n",
+    "\n",
+    "    idle_wires = list(dag.idle_wires())\n",
+    "    for w in idle_wires:\n",
+    "        dag._remove_idle_wire(w)\n",
+    "        dag.qubits.remove(w)\n",
+    "\n",
+    "    dag.qregs = OrderedDict()\n",
+    "\n",
+    "    return dag_to_circuit(dag)\n",
+    "\n",
+    "circs = vqe._circuit_sampler._transpiled_circ_cache\n",
+    "circs = [remove_idle_qwires(circ) for circ in circs]\n",
+    "circ = circs[0]"
```

Comment:
It feels a bit indirect to do a list comprehension over `circs`, but then to only keep one at the end?

Reply:
It was meant to offer the user a handle to both circuits if they wish to visualize them.